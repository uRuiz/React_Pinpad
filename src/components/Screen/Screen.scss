.input-container {
  display: flex;
  justify-content: center;

  input{
    margin-bottom: 32px;
    padding: 16px;
    color: #333;
    background-color: #eee;
    border: none;
    border-radius: 10px;
    text-align: center;
    font-size: 1.5rem;
    width: 65%;
    box-shadow: inset 0 6px #999;

    &.input__ok {
      background-color: rgb(150, 230, 150);
    }
    
    &.input__error {
      background-color: rgb(225, 185, 55);
    }

    &.input__locked {
      background-color: rgb(255, 0, 0);
    }
  }

}

