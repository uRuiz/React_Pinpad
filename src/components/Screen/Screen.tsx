import './Screen.scss';

const SCREEN_MESSAGES = ['OK', 'ERROR', 'LOCKED']

interface iScreenPropsÂ {
  password: string;
}

export const Screen = ({password}: iScreenProps) => {
  const regex = /[\s\S](?!$)/g;
  const encryptedPassword = (SCREEN_MESSAGES.includes(password)) ? password : password.replace(regex, '*');

  return (
    <div className='input-container'>
      <input type='text' disabled value={encryptedPassword}/>
    </div>
  )
}