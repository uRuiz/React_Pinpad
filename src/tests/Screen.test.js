import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import { Screen } from '../components/Screen/Screen';

describe('Tests in <Screen />', () => {
  
  test('It should show <Screen /> properly', () => {
    const password = '2012'
    const wrapper = shallow(<Screen password={password}/>);

    expect(wrapper).toMatchSnapshot();
  });

  test('It should show the password send by props', () => {    
    const password = 'ERROR';
    const wrapper = shallow(<Screen password={password}/>);
    
    expect(wrapper).toMatchSnapshot();
  });

  test('It should show none message', () => {    
    const password = '';
    const wrapper = shallow(<Screen password={password}/>);
    const screenMessage = wrapper.find('input').text()

    expect(screenMessage).toBe(password);
  });
})
