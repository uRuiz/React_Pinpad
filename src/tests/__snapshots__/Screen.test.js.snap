// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <Screen /> It should show <Screen /> properly 1`] = `
<div
  className="input-container"
>
  <input
    className=""
    disabled={true}
    type="text"
    value="***2"
  />
</div>
`;

exports[`Tests in <Screen /> It should show the password send by props 1`] = `
<div
  className="input-container"
>
  <input
    className="input__error"
    disabled={true}
    type="text"
    value="ERROR"
  />
</div>
`;
